tt_metal/impl/dispatch/kernels/command_queue_producer.hpp

```diff
@@ -57,6 +71,49 @@ void issue_queue_pop_front(uint32_t cmd_size_B) {
     notify_host_of_issue_queue_read_pointer();
 }
 
+FORCE_INLINE
+void program_local_cb(uint32_t data_section_addr, uint32_t num_pages, uint32_t page_size, uint32_t cb_size) {
+    uint32_t cb_id = 0;
+    uint32_t fifo_addr = data_section_addr >> 4;
+    uint32_t fifo_limit = fifo_addr + (cb_size >> 4);
+    cb_interface[cb_id].fifo_limit = fifo_limit;  // to check if we need to wrap
+    cb_interface[cb_id].fifo_wr_ptr = fifo_addr;
+    cb_interface[cb_id].fifo_rd_ptr = fifo_addr;
+    cb_interface[cb_id].fifo_size = cb_size >> 4;
+    cb_interface[cb_id].tiles_acked = 0;
+    cb_interface[cb_id].tiles_received = 0;
+    cb_interface[cb_id].fifo_num_pages = num_pages;
+    cb_interface[cb_id].fifo_page_size = page_size >> 4;
+}
+
+template <uint32_t consumer_cmd_base_address, uint32_t consumer_data_buffer_size>
+FORCE_INLINE void program_consumer_cb(bool db_buf_switch, uint64_t consumer_noc_encoding, uint32_t num_pages, uint32_t page_size, uint32_t cb_size) {
+    /*
+        This API programs the double-buffered CB space of the consumer. This API should be called
+        before notifying the consumer that data is available.
+    */
+    uint32_t acked_addr = get_db_cb_ack_addr(db_buf_switch);
```

Comment:
we need to clean this up - later perhaps, as it is spreading.  much better to have a single pointer calculation and offsets.

Reply:
I might clean this up in my PR with the corresponding eth tunneling. I think what I'll do is make an L1 struct and use that for the ethernet cores, while leaving these function getters in for now. We can do a separate cleanup PR that updates the tensix cq consumers/producers too.